core/chains/evm/logpoller/log_poller.go

```diff
@@ -778,7 +807,10 @@ func (lp *logPoller) getCurrentBlockMaybeHandleReorg(ctx context.Context, curren
 // conditions this would be equal to lastProcessed.BlockNumber + 1.
 func (lp *logPoller) PollAndSaveLogs(ctx context.Context, currentBlockNumber int64) {
 	lp.lggr.Debugw("Polling for logs", "currentBlockNumber", currentBlockNumber)
-	latestBlock, err := lp.ec.HeadByNumber(ctx, nil)
+	// If finality is enabled, it's critical to use the finalityDepth returned here, instead of getting one from logPoller.
+	// This is defensive approach to make sure that finalityDepth is not updated by any other Goroutine, during the execution of this function.
```

Comment:
Exactly, this is a defensive approach to increase multithreading safety. The first layer uses channels to avoid multithreading writes to this variable, and the second layer uses local variables instead of accessing it from the field every time. Also, it gives some perf boost, as we don't need to lock/unlock so often.

Reply:
I made it more explicit by rewording comment